app/views/recipes/show.html.erb

```diff
@@ -0,0 +1,59 @@
+<div class="container">
+  <div class="upper_section">
+    <div class="recipe_heading text-center">
+      <h2><%=@recipe.name %></h2>
+    </div>
+    <div class="recipe_details">
+      <div class="d-flex justify-content-between align-items-center">
+        <p class="m-0"><%= "Preparation Time: #{@recipe.preparation_time} hour"%></p>
+         <% if !current_user.nil? && current_user.id == @recipe.user_id %>
+        <%= form_for @recipe do |f| %>
+          <div class="custom-control custom-switch">
+            <%= f.check_box :public, id: "visibility-toggle", class: "custom-control-input" %>
+            <%= f.label :public, class: "custom-control-label", for: "visibility-toggle" %>
+          </div>
+          
+          <%= f.submit "Update Visibility", class: "btn btn-primary" %>
+          <%end%>
+        <% end %>
+      </div>
+      <p><%= "Cooking Time: #{@recipe.cooking_time} hour"%></p>
+      <p><%= "Steps to prepare: #{@recipe.description}"%></p>
+    </div>
+  </div>
+   <% if !current_user.nil? && current_user.id == @recipe.user_id %>
+<div class="buttons mt-5" style='display:flex; justify-content: space-between;'>
+  <%= link_to 'Add Ingredient', new_recipe_recipe_food_path(@recipe.id), class: 'btn btn-primary' %>
+</div>
+<%end%>
+  <div class="table">
+    <table class="table table-striped">
+      <thead>
+        <tr>
+          <th>Food</th>
+          <th>Quantity</th>
+          <th>Value</th>
+          <th>Actions</th>
+        </tr>
+      </thead>
+      <tbody>
+          <% @recipe.foods.each do  |food| %>
+            <tr>
+              <td><%= food.name %></td>
+              <%@quantity = RecipeFood.find_by(food_id: food.id).quantity%>
+              <td><%= @quantity %></td>
+              <td><%= "$ #{food.price * @quantity}" %></td>
+              <% if !current_user.nil? && current_user.id == @recipe.user_id %>
+                <td><%= button_to "remove", recipe_recipe_food_path(@recipe.id, food.id), method: :delete, class:'btn btn-danger' %></td>
+              <% end %>
+            </tr>
+          <% end %>
+        </tbody>
+    </table>
+  </div>
```

Comment:
Nice job so far. However, the recipe details show page needs a little adjustment in order to match the design. 
It is missing the `Generate shopping List` button as shown in the design. You can consider adding it to the left side and ensure that clicking on it will generate the shopping list takes you to the url `/general_shopping_list`.
Kindly consider putting the `Add ingredient` button to the right side just as in the design. :pray: 

### Your design 

![Screenshot from 2023-07-06 22-58-48](https://github.com/imranhossainemi/Recipe_ror_app/assets/65233860/691e2bbb-5357-408a-8d4c-6c42088f6f5c)


### Actual design 

![Screenshot from 2023-07-06 22-58-54](https://github.com/imranhossainemi/Recipe_ror_app/assets/65233860/92a6e0f6-87bb-4b7b-8d1a-005d22dceeb1)

Reply:
@jerryowusu We are people of 2 not 3 why we need that button?

Reply:
@imranhossainemi, I currently do not have the project requirements now, but if you look at it carefully you will see that for people of two you are suppose to implement add and implement this button too.
As seen on the design, that button is suppose to be on the left with the other one on the right.